docs/forgedev/index.md

```diff
@@ -0,0 +1,91 @@
+
+Forking the repository is necessary since not every GitHub user can have free access to every repository. Instead, you create a copy of the original repository to later contribute your changes via a so called Pull Request, which you will learn more about later.
+
+### Cloning
+
+After forking the repository, it is time to get local access to it and to actually make some changes. For this, you need to clone the repository onto your local machine.
+
+Using your favourite git client, simply clone your fork into a directory of your choice. As general example, here is a command line snippet that should work on all correctly configured systems and clones the repository into a directory called "Forge" under the current directory (note that you have to replace `<User>` with your username):
+
+```git clone https://github.com/<User>/MinecraftForge Forge```
+
+# Checking out the Correct Branch
+
+Forking and cloning the repository are the only mandatory steps to develop for Forge. However, to ease the process of creating Pull Requests for you, it is best to work with branches.
+
+It is recommended to create and check out a branch for each PR you plan to submit. This way, you can always keep around the latest changes of Forge for new PRs while you still work on older patches.
+
+After completing this step, you're ready to go and set up your development environment.
+
+Setting Up the Environment
+--------------------------
+
+Depending on your favourite IDE, there is a different set of recommended steps you have to follow to successfully set up a development environment.
+
+### IntelliJ IDEA
+
+JetBrains' flagship IDE comes with great integrated support for [Gradle](https://www.gradle.org), Forge's build system of choice. Due to some peculiarities of Minecraft mod development, however, there are some additional steps required to get everything to work properly.
+
+If you're more of a visual person, our community's very own cpw has uploaded [a video](https://www.youtube.com/watch?v=yanCpy8p2ZE) explaining very similar steps which will also lead to a working setup.
+
+!!! Note
+    These steps will only work reliably from IDEA version 2016 onwards. Older versions didn't have the appropriate Gradle support and did not support Forge development workspaces.
+
+ 1. Import Forge's `build.gradle` as an IDEA project. For this, simply click `File -> Open`, then navigate to your fork's clone and select the `build.gradle` file.
+    If a dialog pops up, select "Open as Project".
+ 2. In the wizard that follows, make sure that "Create separate module per source set" is checked and that the "Use default gradle wrapper" option is active. Confirm the dialog.
+ 3. After IDEA is done importing the project and indexing the files, open the Gradle sidebar on the right hand side of your screen
+ 4. Open the "forge" project tree, select "Tasks", then "forgegradle" and right click the "Create Forge [setup]" option
```

Comment:
I would suggest to run `./gradlew setupForge` in command line. It is way slower to execute in idea.